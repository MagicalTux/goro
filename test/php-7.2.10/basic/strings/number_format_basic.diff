     *** Testing number_format() : basic functionality ***
     
     -- number_format tests.....default --
004- string(5) "1,235"
005- string(6) "-1,235"
004+ string(5) "1,234"
005+ string(6) "-1,234"
     string(10) "12,346,578"
007- string(11) "-12,345,679"
008- string(11) "305,450,479"
009- string(11) "402,653,183"
007+ string(11) "-12,345,678"
008+ string(8) "35,45479"
009+ string(10) "42,653,183"
     string(11) "123,456,789"
     string(3) "123"
     string(3) "123"
013- string(1) "0"
014- string(1) "1"
015- string(1) "0"
013+ panic: runtime error: integer divide by zero
     
017- -- number_format tests.....with two dp --
018- string(8) "1,234.57"
019- string(9) "-1,234.57"
020- string(13) "12,346,578.00"
021- string(14) "-12,345,678.90"
022- string(14) "305,450,479.00"
023- string(14) "402,653,183.00"
024- string(14) "123,456,789.00"
025- string(6) "123.46"
026- string(6) "123.46"
027- string(4) "0.00"
028- string(4) "1.00"
029- string(4) "0.00"
030- 
031- -- number_format tests.....English format --
032- string(8) "1 234.57"
033- string(9) "-1 234.57"
034- string(13) "12 346 578.00"
035- string(14) "-12 345 678.90"
036- string(14) "305 450 479.00"
037- string(14) "402 653 183.00"
038- string(14) "123 456 789.00"
039- string(6) "123.46"
040- string(6) "123.46"
041- string(4) "0.00"
042- string(4) "1.00"
043- string(4) "0.00"
044- 
045- -- number_format tests.....French format --
046- string(8) "1 234,57"
047- string(9) "-1 234,57"
048- string(13) "12 346 578,00"
049- string(14) "-12 345 678,90"
050- string(14) "305 450 479,00"
051- string(14) "402 653 183,00"
052- string(14) "123 456 789,00"
053- string(6) "123,46"
054- string(6) "123,46"
055- string(4) "0,00"
056- string(4) "1,00"
057- string(4) "0,00"
058- ===DONE===
015+ goroutine 1 [running]:
016+ github.com/MagicalTux/goro/ext/standard.fncStrNumberFormat({0x8bf4b0, 0xc000346f80}, {0xc0001288b0, 0x1, 0x1})
017+ 	/home/ronald/proj/goro/ext/standard/strings.go:588 +0x7ea
018+ github.com/MagicalTux/goro/core/phpctx.(*ExtFunction).Call(0xc0001ac450, {0x8bf4b0, 0xc000346f80}, {0xc0001288b0, 0x1, 0x1})
019+ 	/home/ronald/proj/goro/core/phpctx/ext.go:33 +0x9d
020+ github.com/MagicalTux/goro/core/phpctx.(*Global).CallZVal(0xc000274000, {0x8bf388, 0xc000274000}, {0x8bd070, 0xc0001ac450}, {0xc0001288b0, 0x1, 0x1}, {0xc00033d2f0, 0x1, ...})
021+ 	/home/ronald/proj/goro/core/phpctx/global-call.go:84 +0x8a6
022+ github.com/MagicalTux/goro/core/phpctx.(*Global).Call(0xc000274000, {0x8bf388, 0xc000274000}, {0x8bd070, 0xc0001ac450}, {0xc00033c1f0, 0x1, 0x1}, {0xc00033d2f0, 0x1, ...})
023+ 	/home/ronald/proj/goro/core/phpctx/global-call.go:18 +0x32c
024+ github.com/MagicalTux/goro/core/compiler.(*runnableFunctionCallRef).Run(0xc000322810, {0x8bf388, 0xc000274000})
025+ 	/home/ronald/proj/goro/core/compiler/compile-func.go:160 +0x1590
026+ github.com/MagicalTux/goro/core/compiler.(*runOperator).Run(0xc000344000, {0x8bf388, 0xc000274000})
027+ 	/home/ronald/proj/goro/core/compiler/run-operator.go:195 +0x3ea
028+ github.com/MagicalTux/goro/core/phpv.Runnables.Run({0xc0002f3ba0, 0x2, 0x2}, {0x8bf388, 0xc000274000})
029+ 	/home/ronald/proj/goro/core/phpv/run.go:9 +0x103
030+ github.com/MagicalTux/goro/core/compiler.(*runnableFor).Run(0xc0000adec0, {0x8bf388, 0xc000274000})
031+ 	/home/ronald/proj/goro/core/compiler/compile-for.go:86 +0x2fe
032+ github.com/MagicalTux/goro/core/phpv.Runnables.Run({0xc00034e000, 0xb, 0x10}, {0x8bf388, 0xc000274000})
033+ 	/home/ronald/proj/goro/core/phpv/run.go:9 +0x103
034+ github.com/MagicalTux/goro/core/phpctx.(*Global).Require(0xc000274000, {0x8bf388, 0xc000274000}, {0x7ffce0846718, 0x4c})
035+ 	/home/ronald/proj/goro/core/phpctx/global-inc.go:116 +0x4da
036+ github.com/MagicalTux/goro/core/phpctx.(*Global).RunFile(0xc000274000, {0x7ffce0846718, 0x4c})
037+ 	/home/ronald/proj/goro/core/phpctx/global.go:246 +0x7f
038+ main.main()
039+ 	/home/ronald/proj/goro/sapi/php-cli/main.go:60 +0x6a9
