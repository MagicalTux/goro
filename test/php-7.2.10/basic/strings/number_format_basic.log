
---- EXPECTED OUTPUT
*** Testing number_format() : basic functionality ***

-- number_format tests.....default --
string(5) "1,235"
string(6) "-1,235"
string(10) "12,346,578"
string(11) "-12,345,679"
string(11) "305,450,479"
string(11) "402,653,183"
string(11) "123,456,789"
string(3) "123"
string(3) "123"
string(1) "0"
string(1) "1"
string(1) "0"

-- number_format tests.....with two dp --
string(8) "1,234.57"
string(9) "-1,234.57"
string(13) "12,346,578.00"
string(14) "-12,345,678.90"
string(14) "305,450,479.00"
string(14) "402,653,183.00"
string(14) "123,456,789.00"
string(6) "123.46"
string(6) "123.46"
string(4) "0.00"
string(4) "1.00"
string(4) "0.00"

-- number_format tests.....English format --
string(8) "1 234.57"
string(9) "-1 234.57"
string(13) "12 346 578.00"
string(14) "-12 345 678.90"
string(14) "305 450 479.00"
string(14) "402 653 183.00"
string(14) "123 456 789.00"
string(6) "123.46"
string(6) "123.46"
string(4) "0.00"
string(4) "1.00"
string(4) "0.00"

-- number_format tests.....French format --
string(8) "1 234,57"
string(9) "-1 234,57"
string(13) "12 346 578,00"
string(14) "-12 345 678,90"
string(14) "305 450 479,00"
string(14) "402 653 183,00"
string(14) "123 456 789,00"
string(6) "123,46"
string(6) "123,46"
string(4) "0,00"
string(4) "1,00"
string(4) "0,00"
===DONE===
---- ACTUAL OUTPUT
*** Testing number_format() : basic functionality ***

-- number_format tests.....default --
string(5) "1,234"
string(6) "-1,234"
string(10) "12,346,578"
string(11) "-12,345,678"
string(8) "35,45479"
string(10) "42,653,183"
string(11) "123,456,789"
string(3) "123"
string(3) "123"
panic: runtime error: integer divide by zero

goroutine 1 [running]:
github.com/MagicalTux/goro/ext/standard.fncStrNumberFormat({0x8bf4b0, 0xc000346f80}, {0xc0001288b0, 0x1, 0x1})
	/home/ronald/proj/goro/ext/standard/strings.go:588 +0x7ea
github.com/MagicalTux/goro/core/phpctx.(*ExtFunction).Call(0xc0001ac450, {0x8bf4b0, 0xc000346f80}, {0xc0001288b0, 0x1, 0x1})
	/home/ronald/proj/goro/core/phpctx/ext.go:33 +0x9d
github.com/MagicalTux/goro/core/phpctx.(*Global).CallZVal(0xc000274000, {0x8bf388, 0xc000274000}, {0x8bd070, 0xc0001ac450}, {0xc0001288b0, 0x1, 0x1}, {0xc00033d2f0, 0x1, ...})
	/home/ronald/proj/goro/core/phpctx/global-call.go:84 +0x8a6
github.com/MagicalTux/goro/core/phpctx.(*Global).Call(0xc000274000, {0x8bf388, 0xc000274000}, {0x8bd070, 0xc0001ac450}, {0xc00033c1f0, 0x1, 0x1}, {0xc00033d2f0, 0x1, ...})
	/home/ronald/proj/goro/core/phpctx/global-call.go:18 +0x32c
github.com/MagicalTux/goro/core/compiler.(*runnableFunctionCallRef).Run(0xc000322810, {0x8bf388, 0xc000274000})
	/home/ronald/proj/goro/core/compiler/compile-func.go:160 +0x1590
github.com/MagicalTux/goro/core/compiler.(*runOperator).Run(0xc000344000, {0x8bf388, 0xc000274000})
	/home/ronald/proj/goro/core/compiler/run-operator.go:195 +0x3ea
github.com/MagicalTux/goro/core/phpv.Runnables.Run({0xc0002f3ba0, 0x2, 0x2}, {0x8bf388, 0xc000274000})
	/home/ronald/proj/goro/core/phpv/run.go:9 +0x103
github.com/MagicalTux/goro/core/compiler.(*runnableFor).Run(0xc0000adec0, {0x8bf388, 0xc000274000})
	/home/ronald/proj/goro/core/compiler/compile-for.go:86 +0x2fe
github.com/MagicalTux/goro/core/phpv.Runnables.Run({0xc00034e000, 0xb, 0x10}, {0x8bf388, 0xc000274000})
	/home/ronald/proj/goro/core/phpv/run.go:9 +0x103
github.com/MagicalTux/goro/core/phpctx.(*Global).Require(0xc000274000, {0x8bf388, 0xc000274000}, {0x7ffce0846718, 0x4c})
	/home/ronald/proj/goro/core/phpctx/global-inc.go:116 +0x4da
github.com/MagicalTux/goro/core/phpctx.(*Global).RunFile(0xc000274000, {0x7ffce0846718, 0x4c})
	/home/ronald/proj/goro/core/phpctx/global.go:246 +0x7f
main.main()
	/home/ronald/proj/goro/sapi/php-cli/main.go:60 +0x6a9
---- FAILED
