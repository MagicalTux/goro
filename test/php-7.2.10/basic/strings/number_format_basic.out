*** Testing number_format() : basic functionality ***

-- number_format tests.....default --
string(5) "1,234"
string(6) "-1,234"
string(10) "12,346,578"
string(11) "-12,345,678"
string(8) "35,45479"
string(10) "42,653,183"
string(11) "123,456,789"
string(3) "123"
string(3) "123"
panic: runtime error: integer divide by zero

goroutine 1 [running]:
github.com/MagicalTux/goro/ext/standard.fncStrNumberFormat({0x8bf4b0, 0xc000346f80}, {0xc0001288b0, 0x1, 0x1})
	/home/ronald/proj/goro/ext/standard/strings.go:588 +0x7ea
github.com/MagicalTux/goro/core/phpctx.(*ExtFunction).Call(0xc0001ac450, {0x8bf4b0, 0xc000346f80}, {0xc0001288b0, 0x1, 0x1})
	/home/ronald/proj/goro/core/phpctx/ext.go:33 +0x9d
github.com/MagicalTux/goro/core/phpctx.(*Global).CallZVal(0xc000274000, {0x8bf388, 0xc000274000}, {0x8bd070, 0xc0001ac450}, {0xc0001288b0, 0x1, 0x1}, {0xc00033d2f0, 0x1, ...})
	/home/ronald/proj/goro/core/phpctx/global-call.go:84 +0x8a6
github.com/MagicalTux/goro/core/phpctx.(*Global).Call(0xc000274000, {0x8bf388, 0xc000274000}, {0x8bd070, 0xc0001ac450}, {0xc00033c1f0, 0x1, 0x1}, {0xc00033d2f0, 0x1, ...})
	/home/ronald/proj/goro/core/phpctx/global-call.go:18 +0x32c
github.com/MagicalTux/goro/core/compiler.(*runnableFunctionCallRef).Run(0xc000322810, {0x8bf388, 0xc000274000})
	/home/ronald/proj/goro/core/compiler/compile-func.go:160 +0x1590
github.com/MagicalTux/goro/core/compiler.(*runOperator).Run(0xc000344000, {0x8bf388, 0xc000274000})
	/home/ronald/proj/goro/core/compiler/run-operator.go:195 +0x3ea
github.com/MagicalTux/goro/core/phpv.Runnables.Run({0xc0002f3ba0, 0x2, 0x2}, {0x8bf388, 0xc000274000})
	/home/ronald/proj/goro/core/phpv/run.go:9 +0x103
github.com/MagicalTux/goro/core/compiler.(*runnableFor).Run(0xc0000adec0, {0x8bf388, 0xc000274000})
	/home/ronald/proj/goro/core/compiler/compile-for.go:86 +0x2fe
github.com/MagicalTux/goro/core/phpv.Runnables.Run({0xc00034e000, 0xb, 0x10}, {0x8bf388, 0xc000274000})
	/home/ronald/proj/goro/core/phpv/run.go:9 +0x103
github.com/MagicalTux/goro/core/phpctx.(*Global).Require(0xc000274000, {0x8bf388, 0xc000274000}, {0x7ffce0846718, 0x4c})
	/home/ronald/proj/goro/core/phpctx/global-inc.go:116 +0x4da
github.com/MagicalTux/goro/core/phpctx.(*Global).RunFile(0xc000274000, {0x7ffce0846718, 0x4c})
	/home/ronald/proj/goro/core/phpctx/global.go:246 +0x7f
main.main()
	/home/ronald/proj/goro/sapi/php-cli/main.go:60 +0x6a9